src/services/CertificateGenerationService.ts

```diff
@@ -1,898 +1,900 @@
-   */
-  public async getTrailerRegistrationObject(vin: string, make: string) {
-    const config: IInvokeConfig = this.config.getInvokeConfig();
-    const invokeParams: any = {
-      FunctionName: config.functions.trailerRegistration.name,
-      InvocationType: "RequestResponse",
-      LogType: "Tail",
-      Payload: JSON.stringify({
-        httpMethod: "GET",
-        path: `/v1/trailers/${vin}`,
-        pathParameters: {
-          proxy: `/v1/trailers`,
-        },
-        queryStringParameters: {
-          make,
-        },
-      }),
-    };
-    const response = await this.lambdaClient.invoke(invokeParams);
-    try {
-      if (!response.Payload || response.Payload === "") {
-        throw new HTTPError(
-          500,
-          `${ERRORS.LAMBDA_INVOCATION_ERROR} ${response.StatusCode} ${ERRORS.EMPTY_PAYLOAD}`
-        );
-      }
-      const payload: any = JSON.parse(response.Payload as string);
-      if (payload.statusCode === 404) {
-        console.debug(`vinOrChassisWithMake not found ${vin + make}`);
-        return { Trn: undefined, IsTrailer: true };
-      }
-      if (payload.statusCode >= 400) {
-        throw new HTTPError(
-          500,
-          `${ERRORS.LAMBDA_INVOCATION_ERROR} ${payload.statusCode} ${payload.body}`
-        );
-      }
-      const trailerRegistration = JSON.parse(
-        payload.body
-      ) as ITrailerRegistration;
-      return { Trn: trailerRegistration.trn, IsTrailer: true };
-    } catch (err) {
-      console.error(
-        `Error on fetching vinOrChassisWithMake ${vin + make}`,
-        err
-      );
-      throw err;
+    /**
+     * Used to get a singular whole technical record.
+     * @param systemNumber
+     * @param createdTimestamp
+     */
+    public callGetTechRecords = async <T extends TechRecordGet["techRecord_vehicleType"]>(systemNumber: string, createdTimestamp: string): Promise<TechRecordType<T> | undefined> => {
+        const config: IInvokeConfig = this.config.getInvokeConfig();
+        const invokeParams: InvocationRequest = {
+            FunctionName: config.functions.techRecords.name,
+            InvocationType: "RequestResponse",
+            LogType: "Tail",
+            Payload: JSON.stringify({
+                httpMethod: "GET",
+                path: `/v3/technical-records/${systemNumber}/${createdTimestamp}`,
+                pathParameters: {
+                    systemNumber,
+                    createdTimestamp
+                }
+            }),
+        };
+        return await this.lambdaClient.invoke(invokeParams)
+            .then(async (response) => {
+                try {
+                    const res = await this.lambdaClient.validateInvocationResponse(response);
+                    return JSON.parse(res.body);
+                } catch (e) {
+                    console.log("in get tech record catch block");
+                    console.log(e);
+                    return undefined;
+                }
+            });
```

Comment:
Same comment as above regarding the use of both `.then` and `async/await` syntax